### PR TITLE
MacOS builds for pipelines

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Adds macOS Monterey (x86_64 and ARM64) builds for ocaml-ci to all pipelines.

Currently we support 2 OCaml compilers 4.14 and 5.0 on both architectures. We could add more if there was sufficient interest and value in testing them.

### CODE REVIEW

File: service/conf.ml [hunk 1]
```diff
@@ -63,14 +63,58 @@ type platform = {
...
+    {
+      label = "macos-homebrew";
+      builder = Builders.local;
+      pool = "macos-x86_64";
+      distro = "macos-homebrew";
+      ocaml_version = OV.Releases.v5_0;
+      arch = `X86_64;
+      opam_version = `V2_1;
+    };
+    (* Apple Silicon *)
```
Comment (tmcgilchrist) [inline service/conf.ml:89]: Even though we can support macOS arm64, we only have two machines available. @mtelvers

**Review by novemberkilo — APPROVED** [review]
👍🏾 @tmcgilchrist 

I don't know this area of the code well enough to add much to the review. I used it as an opportunity to learn ... thanks for that. ~~I'm curious to understand what the next steps are~~ (this PR makes it pretty clear that we are starting out with some hard-coded configuration, and with pretend MacOS docker containers).

Edit: I see the tracking ticket now. Sorry.

File: service/conf.ml (hunk 1 above)
Comment (tmcgilchrist) [reply service/conf.ml:89]: Additional macOS arm64 machines will soon be added to the cluster.

Comment (tmcgilchrist) [reply service/conf.ml:89]: Additional macOS arm64 machines are available, I will enable both architectures.

**Pull request merged**